src/rust/engine/logging/src/executor.rs

```diff
@@ -0,0 +1,88 @@
+use futures::Future;
+use std::sync::Arc;
+use tokio::runtime::Runtime;
+
+#[derive(Clone)]
+pub struct Executor {
+  runtime: Arc<Runtime>,
+}
+
+impl Executor {
+  pub fn new() -> Executor {
+    Executor {
+      runtime: Arc::new(
+        Runtime::new().unwrap_or_else(|e| panic!("Could not initialize Runtime: {:?}", e)),
+      ),
+    }
+  }
+
+  ///
+  /// Start running a Future on a tokio Runtime.
+  ///
+  pub fn spawn<F: Future<Item = (), Error = ()> + Send + 'static>(&self, future: F) {
+    // Make sure to copy our (thread-local) logging destination into the task.
+    // When a daemon thread kicks off a future, it should log like a daemon thread (and similarly
+    // for a user-facing thread).
+    let logging_destination = crate::get_destination();
+    self.runtime.executor().spawn(futures::lazy(move || {
+      crate::set_destination(logging_destination);
+      future
+    }))
+  }
+
+  ///
+  /// Run a Future on a tokio Runtime as a new Task. This is useful for blocking tokio tasks,
+  /// because blocking blocks the entire task called blocking, not just the part of it which is
+  /// marked as blocking. This allows finer granularity of marking tasks as blocking.
+  ///
+  /// See https://docs.rs/tokio-threadpool/0.1.15/tokio_threadpool/fn.blocking.html
+  ///
+  pub fn spawn_in_new_task<
```

Comment:
So, my understanding of the word `spawn` is that it _always_ implies a new `Task`, disconnected from the current one.

So is this actually different from the `spawn` above?

Reply:
I _think_ the key difference is that this spawns _but gives you a handle you can wait on_, whereas the `spawn` above spawns and drives completion of the future opaquely... I _think_...

Reply:
Got it. Yep. So maybe a naming nit then.